### ISSUE TITLE
"Two collaborative sessions" warning when one user opens the same file twice, with different schemas, in the same session

### ISSUE THREAD

**Issue by JasonWeill** [jupyterlab/jupyter-collaboration#371]
## Description

When I open a file twice, with two different schemas in the same Lab session with Collaboration enabled, I see a warning: 

> Two collaborative sessions are accessing the file _(name)_ simultaneously. Opening a document with multiple views simultaneously is not supported. Please close one view; otherwise, you might lose some of your progress.

![Screenshot 2024-02-26 at 11 42 20 AM (1)](https://github.com/jupyterlab/jupyter-collaboration/assets/4554746/af1e5ce3-d34f-4a39-aee7-9b83bb42181a)

## Reproduce

1. Create a notebook, `foo.ipynb`. Save it. Leave it open.
2. Right-click on `foo.ipynb` in the file browser. Hover over "Open With", and click "Editor."
3. The notebook opens in what looks like a text file editor. The warning message above appears.

## Expected behavior

Either the open operation fails entirely with a message that the file is already open, or the file opens without a warning of any sort.

Personally, I think it would be acceptable to, when Jupyter Collaboration is installed, reject opening the same file more than once, with different schemas.

## Context

This warning was first seen in #245, when creating a new file.

Opening the same notebook in two separate browser tabs/windows does not cause this error.

**Comment by davidbrochart** [comment]
I don't think this is a bug. It's fine to open the same file twice as long as changes are made on only one view.

**Comment by JasonWeill** [comment]
When opening the same file a second time, is it feasible to detect that the file is already open in a read/write editor, and force the second editor to be read-only?

**Comment by davidbrochart** [comment]
I think that would complicate the picture. I would rather make it possible to collaborate between different views, as I tried to explain in https://github.com/jupyter-server/jupyter_ydoc/issues/15.
